### DISCUSSION TITLE
Recomendações de livros técnicos e de carreira

### DISCUSSION THREAD

**Discussion by camillodev** [frontendbr/forum#1323] [Discussão]
Oi galera !
To querendo criar o hábito de leitura e to querendo indicações de livros tanto técnicos quanto de desenvolvimento pessoal que sejam uteis na vida profissional. 
Já teve uma issue dessa, mas não teve muitas indicações, então to criando essa na esperança que bombe de listinha de livros legais.

**Comment by onlurking** [comment]
Cara, alguns que eu li e achei muito bom:

**Livros Técnicos:**

[CSS in Depth](https://www.manning.com/books/css-in-depth)
Eu ainda estou lendo ele, mas na real, é o melhor livro que eu já topei sobre CSS, ele explica como funciona muitas coisas por debaixo dos panos, se tu meche com frontend é uma leitura indispensável imho.

**Desenvolvimento Profissional:**

[The Clean Coder](https://www.goodreads.com/book/show/10284614-the-clean-coder)
Esse é o livro que eu aprendi sobre como estimar software direito (usando 3 medidas de tempo pra cada feature: tempo pessimista, tempo mediano e tempo ótimo), é um apanhado gigantesco de coisas que o Martin adoraria saber antes de aprender se ferrando na vida.

[Soft Skills, The software developer's life manual](https://www.manning.com/books/soft-skills)
Ele dá varios insights de como fazer networking(e conseguir alguns jobs com isso), criar hábitos saudáveis que podem alavancar sua carreira, problemas que você pode enfrentar como programador (burnout) e muitas outras coisas, porém eu recomendo você pular toda a parte que o autor fala sobre finanças, é o ponto fraco desse livro.

[Pragmatic Thinking and Learning](https://pragprog.com/book/ahptl/pragmatic-thinking-and-learning) 
Um livro sobre como otimizar o aprendizado e como o cérebro funciona, uma das melhores leituras que tive sobre o assunto (e do lado coloco o livro *aprendendo inteligência* do pierluigi piazzi).

[97 Things Every Programmer Should Know](https://leanpub.com/97-Things-Every-Programmer-Should-Know-Extended)
Haha, faz um tempão que eu li esse livro mas eu lembro que tinha muito gold nele, é um apanhando de recomendações de vários programadores, coisas que eles queriam saber antes antes de aprender do jeito difícil.

**ps:** Eu ainda vou editar essa resposta com mais recomendações na parte técnica e o que eu acho.

**Comment by valterbarros** [comment]
https://www.youtube.com/watch?v=wcGGklDfHM4 ve esse vídeo do akita ele fala
sobre o The MM-M

Em qui, 20 de dez de 2018 às 23:20, Diogo Felix <notifications@github.com>
escreveu:

**Comment by felipefialho** [comment]
Linkando: #229 #253 

Além das dicas de livros, sugiro dar uma lida no próprio fórum, em tópicos como: #132 #118 #1259 e etc

**Comment by camillodev** [comment]
Brigadão pelas referências gente. Ja tem muita coisa massa ai, vou começar a ler e comprar os livros. Vlw

**Comment by diegogriep** [comment]
Focando na parte "carreira", recomendo o livro **Como fazer amigos e influenciar pessoas**, do **Dale Carnegie**. São ótimas dicas, entre algumas, conselhos de como se relacionar, de ser uma pessoa com mais empatia.

**Comment by camillodev** [comment]
@diegogriep otima recomendação. Esse livro mudou muito a forma como eu me relacionava com as pessoas. E genial.

**Comment by gustavofsantos** [comment]
Se tu curte matemática, eu li esses aqui e curti bastante:
- [Fantástico Mundo dos Números](https://www.amazon.com.br/fant%C3%A1stico-mundo-dos-n%C3%BAmeros-matem%C3%A1tica/dp/8537815527/ref=sr_1_1?s=books&ie=UTF8&qid=1545417096&sr=1-1&keywords=fantastico+mundo+dos+numeros)
- [The Code Book](https://www.amazon.com.br/Code-Book-Science-Secrecy-Cryptography/dp/0385495323/ref=sr_1_1?s=books&ie=UTF8&qid=1545417138&sr=1-1&keywords=the++code+book)

Também tenho um de inteligência artifical que comprei no início da faculdade, com a primeira bolsa de IC:
- [Inteligência Artificial](https://www.amazon.com.br/Intelig%C3%AAncia-Artificial-Peter-Norvig/dp/8535237011/ref=sr_1_1?s=books&ie=UTF8&qid=1545417472&sr=1-1&keywords=inteligencia+artificial): esse livro é muito (muito, muito) denso, vou colar grau daqui algumas semanas e ainda não consegui ler todo ele

Também curti [O Poder do Hábito](https://www.amazon.com.br/poder-do-h%C3%A1bito-Charles-Duhigg/dp/8539004119/ref=sr_1_1?s=books&ie=UTF8&qid=1545417269&sr=1-1&keywords=o+poder+do+habito), é um pouco repetitivo, mas o conteúdo é muito pra te ajudar a criar bons hábitos. Também tenho o [Linux a Biblia](https://www.amazon.com.br/Linux-B%C3%ADblia-Christopher-Negus/dp/8576087995/ref=sr_1_1?s=books&ie=UTF8&qid=1545417363&sr=1-1&keywords=linux+biblia) pra consulta rápida, mas aprendi muitas coisas sobre Linux nesse livro.

[Esse](https://www.amazon.com.br/Express%C3%B5es-Regulares-Uma-Abordagem-Divertida/dp/8575224743/ref=sr_1_1?ie=UTF8&qid=1545417586&sr=8-1&keywords=express%C3%B5es+regulares) livro do Aurélio é muito bom pra começar a entender expressões regulares, ainda estou lendo ele e cara, vale a pena.

Sem esquecer da série [You Dont Know JS](https://github.com/getify/You-Dont-Know-JS) do Kyle que dispensa comentários, é excelente em todos os volumes e está completo no GitHub. Também curti o [The Nature of Code](https://natureofcode.com/) do Daniel Shiffman, que inclusive tem um canal no [YT](https://www.youtube.com/channel/UCvjgXvBlbQiydffZU7m1_aw) muito legal, o livro também está disponível [nesse repositório no GitHub](https://github.com/shiffman/The-Nature-of-Code), inclusive tu consegue compilar ele e ver as animações bonitinhas

**Comment by HmmerHead** [comment]
Poder do hábito é um ótimo livro. Vale muito a pena ler, e começar o ano com força total.

**Comment by lu-zen** [comment]
Algorithms 4th Edition - Sedgewick & Wayne
Recente, explica algorítmos de forma pragmática.
